### ISSUE TITLE
jsdoc/require-returns-check: False positive for overloads

### ISSUE THREAD

**Issue by ilyub** [gajus/eslint-plugin-jsdoc#907]
Sample:
```ts
  /**
   * Description.
   *
   * @returns Result.
   */
  function f(): string;

  function f(): string {
    return "Result";
  }
```

## Expected behavior
No warnings.

## Actual behavior
Getting warning:
JSDoc @returns declaration present but return expression not available in function

If this plugin is not using typescript then at least it might ignore overloads with additional option.
